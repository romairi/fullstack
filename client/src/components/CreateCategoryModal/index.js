import React from "react";
import './index.scss';
import Button from "@material-ui/core/Button";
import Modal from "@material-ui/core/Modal";
import Backdrop from "@material-ui/core/Backdrop/Backdrop";
import Fade from "@material-ui/core/Fade/Fade";
import TextField from "@material-ui/core/TextField/TextField";


function CreateCategoryModal({onAddCategoryClicked, isModalOpen, setModalOpen}) {
    const [categoryParam, setCategoryParams] = React.useState('');

    const onCategoryNameChange = (event) => {
        setCategoryParams(event.target.value);

    };

    const onAddButtonClicked = () => {
        onAddCategoryClicked && onAddCategoryClicked(categoryParam);
        setCategoryParams('');
    };

    const closeModal = () => setModalOpen(false);

    return (
            <Modal
                className="modal"
                aria-labelledby="transition-modal-title"
                aria-describedby="transition-modal-description"
                open={isModalOpen}
                onClose={() => setModalOpen(false)}
                closeAfterTransition
                BackdropComponent={Backdrop}
                BackdropProps={{
                    timeout: 500,
                }}
            >
                <Fade in={isModalOpen}>
                    <div className="modal_box">
                        <h3 className="title_modal_box">Add new category</h3>
                        <div className="add_category">
                            <TextField
                                className="category_textField"
                                type="text"
                                label="Name category"
                                onChange={onCategoryNameChange}
                                value={categoryParam}
                                variant="outlined"
                            />
                        </div>
                        <div className="footer_modal">
                            <Button
                                className="btn_close"
                                variant="contained"
                                size="large"
                                onClick={closeModal}>
                                Close
                            </Button>
                            <Button
                                className="btn_add"
                                variant="contained"
                                size="large"
                                onClick={onAddButtonClicked}>
                                Add
                            </Button>
                        </div>
                    </div>
                </Fade>
            </Modal>
    );
}

export default CreateCategoryModal;
