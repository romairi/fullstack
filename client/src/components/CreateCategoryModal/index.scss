@import "../../scss/variables";

.modal {
  display: flex;
  align-items: center;
  justify-content: center;

  .modal_box {
    width: 500px;
    height: 300px;
    border: 2px solid #000;
    box-shadow: 3px 3px 5px 4px #beb5b5;
    padding: 30px;
    background: white;

    .title_modal_box {
      text-align: center;
      color: $primary_color;
      font-family: 'Dancing Script', cursive;
    }

    .add_category {
      display: flex;
      justify-content: center;
      margin-top: 40px;

      .category_textField {
        width: 95%;

        .MuiInputLabel-formControl {
          font-family: 'Dancing Script', cursive;
        }
      }

    }

    .footer_modal {
      display: flex;
      justify-content: center;
      margin-top: 50px;

      .btn_add {
        color: $color_white;
        background-color: $primary_color;
        font-family: 'Dancing Script', cursive;
        padding-right: 30px;
        padding-left: 30px;

        &:hover {
          background-color: $primary_hover_color;
        }
      }

      .btn_close {
        font-family: 'Dancing Script', cursive;
        margin-right: 40px;
      }
    }
  }
}
