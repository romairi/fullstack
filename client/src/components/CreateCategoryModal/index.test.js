import React from 'react';
import renderer from 'react-test-renderer';
import CreateCategoryModal from './index';


jest.mock('@material-ui/core/Button', () => () => <div className='Button'/>);
jest.mock('@material-ui/core/Modal', () => () => <div className='Modal'/>);
jest.mock('@material-ui/core/Backdrop/Backdrop', () => () => <div className='Backdrop'/>);
jest.mock('@material-ui/core/Fade/Fade', () => () => <div className='Fade'/>);
jest.mock('@material-ui/core/TextField/TextField', () => () => <div className='TextField'/>);

describe('CreateCategoryModal', () => {
    describe('render', () => {
        it('should render the component - with open modal', () => {
            const component = renderer.create(
                <CreateCategoryModal/>
            );
            expect(component.toJSON()).toMatchSnapshot();
        });
    });

});
