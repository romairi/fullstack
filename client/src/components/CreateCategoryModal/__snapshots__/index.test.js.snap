// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCategoryModal render should render the component - with open modal 1`] = `
<div
  className="Modal"
/>
`;
